iroh-net/src/discovery/local_swarm_discovery.rs

```diff
@@ -28,7 +59,7 @@ use crate::{
 const N0_LOCAL_SWARM: &str = "iroh.local.swarm";
 
 /// Provenance string
-const PROVENANCE: &str = "local.swarm.discovery";
+const PROVENANCE: &str = "local";
```

Comment:
After reviewing the other discovery services they all had short broad names like "relay" "mainline", "pkarr" etc. But, I don't hate "local.swarm.discovery", since "local" is arguably more broad than the others. I'll change it back.

I initially wanted to `pub` this and do the same for each discovery service. Namely, have a pub const called something like `SOURCE_NAME` so that people could do this trick for each discovery service and so that it would be clearer for a user how to search for each discovery source BUT the dht discovery service has two different sources, so I couldn't use `SOURCE_NAME` for each. Because it required more thought to make sure I wouldn't be breaking the public API a bunch of times, I decided not to do it rn & to try to come up with another solution.

But, I don't hate the idea of just doing it with just this source, since it has, so far, been the only one that users have asked for. I'll change it to `SOURCE_NAME`, add documentation, and make it public. If others complain to us, then we can adjust it in the future.

What do you think?